### ISSUE TITLE
Classes inheriting from other classes, defined in mocked modules, are ignored

### ISSUE THREAD

**Issue by ederag** [sphinx-doc/sphinx#4956]
As advised by @djhoese, this is a reopening of #1880.
Unfortunately, this is not #2174 (solved).
Actually, the docstrings of classes inheriting from other classes in mocked modules are ignored.
A real case is described in https://stackoverflow.com/q/50203452/3565696.
Here is a minimal working example, all files in the same directory
(also attached as [sphinx-mock-bug.zip](https://github.com/sphinx-doc/sphinx/files/1993341/sphinx-mock-bug.zip), for convenience).

Project module `foo.y`:
```
import bar

class FooClass(bar.BarClass):
    """This is myclass"""
    pass
```
`conf.py`:
```
import sys
import os

# The FooClass docstring shows up only if this line is commented out
autodoc_mock_imports = ['bar']

sys.path.insert(0, os.path.abspath('.'))

extensions = [
    'sphinx.ext.autodoc',
]
```
`contents.rst`:
```
foo
===

.. automodule:: foo
    :members:
    :noindex:
```
And finally, the`bar.py` "external" module (only used when the autodoc_mock_imports is commented out):
```
class BarClass():
	pass
```

The command used to build:
`sphinx-build  -b html . _build/html`

To see the file where the dosctring should be visible:
`firefox _build/html/foo.html`

Environment info:
- Linux (openSUSE 42.2)
- Python version: 3.4.6 and 3.6.4
- Sphinx version: 1.7.4
[and same on readthedocs for the real project]

**Comment by tk0miya** [comment]
Thank you for reporting. I'll take a look this later.

**Comment by tk0miya** [comment]
At present, it can't make subclasses of the class on mocked module.
```
from sphinx.ext.autodoc.importer import _MockObject

bar = _MockObject()

class FooClass(bar.BarClass):
    """docstring of FooClass"""
    pass


foo = FooClass()
print(foo)          # => <sphinx.ext.autodoc.importer._MockObject object at 0x1101fc828>
print(foo.__doc__)  # => Used by autodoc_mock_imports.
```

@shimizukawa Do you have any idea for this?

**Comment by tk0miya** [comment]
I just made a #4995 . I hope this resolves the issue.
@ederag Could you try this please?

**Comment by ederag** [comment]
@tk0miya Confirmed, this issue is fixed by #4995. Congratulations !
The geoptics documentation now chokes on signal lines as `signal_set_all_selected = pyqtSignal(bool)`.
It deserves another issue, because commenting out those lines allows the build to complete.

**Comment by tk0miya** [comment]
Thank you for confirmation! I'll merge it later (after fixing CI errors).

**Comment by ederag** [comment]
The `pyqtSignal` issue is very close to #4931. [Comment](https://github.com/sphinx-doc/sphinx/issues/4931#issuecomment-390496070) left there, with a minimal working example.

**Comment by tk0miya** [comment]
Okay, I'll check it in the thread.

**Comment by tk0miya** [comment]
Fixed by #4995
Thank you for reporting :-)

**Comment by ederag** [comment]
Now it is possible to document PyQt5 projects on readthedocs, with [simple configuration tweaks](https://stackoverflow.com/a/50745283/3565696).
Thanks a lot !
